src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,80 @@
+import { Response, NextFunction } from 'express';
+import passport from 'passport';
+
+import { getUserService, createUserService } from '../services/auth.service';
+import { getToken } from '../utils/auth.util';
+
+import { IUser } from '../models/user.model';
+
+export const getUser = async (req: any, res: Response) => {
+	try {
+		const user: IUser | null = await getUserService(req.user._id);
+		return res.status(200).json({ status: 200, data: user });
+	} catch (e) {
+		return res
+			.status(e.status || 500)
+			.json({ status: e.status || 500, message: e.message });
+	}
+};
+
+export const loginUser = async (
+	req: any,
+	res: Response,
+	next: NextFunction
+) => {
+	try {
+		passport.authenticate(
+			'local',
+			{ session: false },
+			(err: any, user: IUser, info: any) => {
+				if (err) return next(err);
+				if (!user)
+					return res.status(401).json({
+						success: false,
+						status: 'Login Unsuccessful!',
+						err: info
+					});
+
+				req.logIn(user, (err: any) => {
+					if (err)
+						return res.status(401).json({
+							success: false,
+							status: 'Login Unsuccessful!',
+							err: err.message
```

Comment:
Would these error messages from auth be easy to understand? Wouldn't it be a better idea to add our own text for the error messages? The 'status' in the return json could also be more specific right?

Reply:
Create an issue for this. 
I think this is another wrapper @Ishar-Menon and team can work on.